grid/crm/contacts/maps/crisp.suma

```diff
@@ -0,0 +1,286 @@
+map Create {
+  set {
+    newProfile = call BuildCrispProfile(input = input)
+  }
+
+  return map error if (!newProfile.email || !(newProfile.person || {}).nickname) {
+    title = "Missing fields"
+    detail = "Crisp requires Email and Name to create new contact"
+  }
+
+  http POST "/v1/website/{parameters.WEBSITE_ID}/people/profile" {
+    security "plugin_tier"
+
+    request "application/json" {
+      headers { "X-Crisp-Tier" = "plugin" }
+      body = newProfile
+    }
+
+    response 201 {
+      set {
+        customPropsResult = call PatchCustomProperties(
+          profileId = body.data.people_id,
+          customProperties = input.customProperties
+        )
+      }
+
+      return map error if (!customPropsResult) {
+        title = "Partially failed"
+        detail = "Contact might have been created without custom properties. Please update the contact for consistency"
+      }
+
+      map result {
+        id = body.data.people_id
+      }
+    }
+
+    response 409 {
+      map error {
+        title = "Contact already exists"
+        detail = (`Contact '${input.email}' already exists. If you with to update it, use 'Update' use case`)
+      }
+    }
+
+    response {
+      error = call HandleCommonErrors(statusCode = statusCode)
+      map error error
+    }
+  }
+}
+
+map Update {
+  set {
+    profileId = input.id || input.email
+    profileUpdate = call BuildCrispProfile(input = input)
+  }
+
+  http PATCH "/v1/website/{parameters.WEBSITE_ID}/people/profile/{profileId}" {
+    security "plugin_tier"
+
+    request "application/json" {
+      headers { "X-Crisp-Tier" = "plugin" }
+      body = profileUpdate
+    }
+
+    response 200 {
+      set {
+        customPropsResult = call PatchCustomProperties(
+          profileId = profileId,
+          customProperties = input.customProperties
+        )
+      }
+
+      return map error if (!customPropsResult) {
+        title = "Partially failed"
+        detail = "Contact might be updated without custom properties. Please update the contact for data consistency"
+      }
+
+      map result {
+        id = profileId
+      }
+    }
+
+    response 409 {
+      map error {
+        title = "Email already exists"
+        detail = (`Email '${input.email}' already exists in another contact.`)
+      }
+    }
+
+    response {
+      error = call HandleCommonErrors(statusCode = statusCode)
+      map error error
+    }
+  }
+}
+
+map Search {
+  OPERATOR_MAP = {
+    "EQ": 'eq',
+    "NEQ": 'neq'
+  }
+
+  search_filter = [{
+    criterion : input.property,
+    query : [input.value],
+    operator : OPERATOR_MAP[input.operator] || input.operator,
+    model: "people"
+  }]
+
+  search_filter_query_param = encodeURIComponent(JSON.stringify(search_filter))
+
+  http GET "/v1/website/{parameters.WEBSITE_ID}/people/profiles?search_filter={search_filter_query_param}" {
+    security "plugin_tier"
+
+    request "application/json" {
+      headers { "X-Crisp-Tier" = "plugin" }
```

Comment:
Is there a problem with using `query { search_filter = search_filter }` here? (note it has to go before the `headers { ... }` slot, as the ordering is fixed right now).

Reply:
Guess I had a blackout.... `query` slot works well :)